src/parser/fragments/object_location.rs

```diff
@@ -10,6 +10,10 @@ use nom::IResult;
 /// **Type ID:** 0x15
 pub struct ObjectLocationFragment {
     pub name_reference: StringReference,
+    /// When used in main zone files, the reference points to a 0x14 Player Info fragment. When used for static (placeable) objects,
+    /// the reference is a string reference (not a fragment reference) and points to a “magic” string.
+    /// It typically contains the name of the object with “_ACTORDEF” appended to the end.
+    pub reference: StringReference, // FIXME: This can be a FragmentRef sometimes, as stated above
```

Comment:
Ah, yeah, that's a bummer. I guess `FragmentRef` only handles the case where we're referencing a fragment using it's string not referencing the string itself. Maybe this requires a new type 😿 . I guess at least we _are_ able to determine which it is based on if the value is postive or negative, right? Maybe we could just create something like:

```
enum FragmentOrStringRef<T> {
  String(StringReference)
  Fragment(FragmentRef<T>)
}
```

And, like `FragmentRef::new`, create the corrrect variant based on if it's positive or negative. In fact you could probably just re-use `FragmentRef::new` and unwrap the `StringReference` it creates if it ends up being a "Name" variant.